aixplain/api/pipeline.py

```diff
@@ -0,0 +1,354 @@
+import uuid
+from dataclasses import dataclass, field, InitVar, asdict
+from typing import List, Tuple
+
+from aixplain.factories.pipeline_factory import PipelineFactory
+
+from .enums import (
+    NodeType,
+    DataType,
+    RouteType,
+    Operation,
+)
+from .base import Node, Link
+from .nodes import (
+    Asset,
+    Decision,
+    Script,
+    Input,
+    Output,
+    Route,
+    Router,
+    Reconstructor,
+    Segmentor,
+)
+
+
+@dataclass
+class Pipeline:
+    nodes: List[Node] = field(default_factory=list)
+    links: List[Link] = field(default_factory=list)
+    instance: InitVar[any] = None
+
+    def add_node(self, node: Node):
+        """
+        Add a node to the current pipeline.
+
+        This method will take care of setting the pipeline instance to the
+        node and setting the node number if it's not set.
+
+        :param node: the node
+        :return: the node
+        """
+        return node.attach(self)
+
+    def add_nodes(self, *nodes: Node) -> List[Node]:
+        """
+        Add multiple nodes to the current pipeline.
+
+        :param nodes: the nodes
+        :return: the nodes
+        """
+        return [self.add_node(node) for node in nodes]
+
+    def add_link(self, link: Link) -> Link:
+        """
+        Add a link to the current pipeline.
+        :param link: the link
+        :return: the link
+        """
+        return link.attach(self)
+
+    def input(self, data: str = None, *args, **kwargs) -> Node:
+        """
+        Shortcut to create an input node for the current pipeline.
+        All params will be passed as keyword arguments to the node
+        constructor.
+
+        `data` is a special convenient parameter that will be uploaded to the
+        aixplain platform and the link will be passed as the input to the node.
+
+        :param data: the data to be uploaded
+        :param args: positional arguments
+        :param kwargs: keyword arguments
+        :return: the node
+        """
+        kwargs["data"] = data
+        return Input(self, *args, **kwargs)
+
+    def asset(self, assetId: str, *args, **kwargs) -> Node:
```

Comment:
Enable the user to also give the instance of a model:

```
assetId: Union[Model, Text]
```

Reply:
This is sth i already planned (was certainly there in the first drafts) but missed to put it back. The same is also valid for Input node for the use of data assets. Let me check this by today. Will update accordingly

Reply:
Actually, indeed it was already there under `instance` param but still needs to be more convenient through. Will work on it.

Reply:
Ok this is done for assets and also Input while handling data from file. But we need to implement data asset, corpus or csv type of data usage which i need to understand how we could achieve that. But as a matter of having the first version we can proceed with as is and leave it as a next improvement.